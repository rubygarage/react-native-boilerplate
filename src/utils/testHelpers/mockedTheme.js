import { keys, includes } from 'ramda';

const theme = {
  type: 'Mocked theme',
  colors: {},
  sizes: {},
  fonts: {},
};

const handler = {
  get: (target, key) => {
    const { type } = target;
    const property = target[key];

    if (typeof property === 'object') {
      property.type = key;
      return new Proxy(property, handler);
    }

    if (includes(type, keys(theme))) {
      return type === 'sizes' ? 0 : `[mocked ${key}]`;
    }

    return Reflect.get(target, key);
  },
};

export default new Proxy(theme, handler);
