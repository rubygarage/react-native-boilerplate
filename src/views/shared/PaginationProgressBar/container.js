import { withTheme } from 'styled-components/native';

import PaginationProgressBarComponent from './component';

export default withTheme(PaginationProgressBarComponent);
