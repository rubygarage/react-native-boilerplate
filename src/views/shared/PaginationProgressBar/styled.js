import styled from 'styled-components/native';

const Wrap = styled.TouchableOpacity`
  width: 100%;
  height: 40px;
  align-items: center;
  justify-content: center;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
