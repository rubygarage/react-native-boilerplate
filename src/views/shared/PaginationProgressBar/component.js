import React from 'react';
import { ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';

import * as S from './styled';

const PaginationProgressBar = ({ isLoading, theme }) => isLoading && (
  <S.Wrap>
    <ActivityIndicator color={theme.colors.blue100} />
  </S.Wrap>
);

PaginationProgressBar.propTypes = {
  theme: PropTypes.shape({ colors: PropTypes.shape().isRequired }).isRequired,
  isLoading: PropTypes.bool,
};

PaginationProgressBar.defaultProps = {
  isLoading: false,
};

export default PaginationProgressBar;
