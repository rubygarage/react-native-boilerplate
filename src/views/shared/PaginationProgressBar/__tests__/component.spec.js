import React from 'react';
import { shallow } from 'enzyme';

import mockedTheme from 'utils/testHelpers/mockedTheme';

import PaginationProgressBar from '../component';

jest.mock('../hook', () => ({
  __esModule: true,
  default: jest.fn(() => mockedTheme),
}));

describe('PaginationProgressBar component', () => {
  const defaultProps = {
    isLoading: true,
  };

  const component = shallow(<PaginationProgressBar {...defaultProps} />);

  it('renders correctly when loading', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when not loading', () => {
    component.setProps({ ...defaultProps, isLoading: false });
    expect(component).toMatchSnapshot();
  });
});
