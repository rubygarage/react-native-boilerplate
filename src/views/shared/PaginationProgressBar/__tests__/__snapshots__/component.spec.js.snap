// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationProgressBar component renders correctly when loading 1`] = `
<Styled(TouchableOpacity)>
  <ActivityIndicator
    animating={true}
    color="[mocked blue100]"
    hidesWhenStopped={true}
    size="small"
  />
</Styled(TouchableOpacity)>
`;

exports[`PaginationProgressBar component renders correctly when not loading 1`] = `""`;
